### PR TITLE
RE-1408 Implement nodepool metrics stack

### PR DESCRIPTION
**Pull request by odyssey4me**
In order to help understand the health of nodepool,
its nodes and a bit about its usage, we implement
the configuration and new systems to allow nodepool
to send metrics to statsd, then push them to graphite,
then graph them in grafana with dashboards defined in
the git tree using yaml configuration.

This combination mirrors the OpenStack-Infra implementation.

The DNS entries have not yet been created as I need the
IP addresses of the instances before I can do so. I'd
like to get some reviews before going ahead and completing
that. Also, the API key for grafana can only be generated
once it's up - so for now I've not defined it. Once the
servers are built I'll generate it, drop it into password
safe and implement the extra variable.

Finally, several new jobs are added here which duplicate
a lot of things. I've registered [RE-1408](https://rpc-openstack.atlassian.net/browse/RE-1472) to follow up on that
and reduce the duplication.

Issue: [RE-1408](https://rpc-openstack.atlassian.net/browse/RE-1408)

### CODE REVIEW

**Comment by odyssey4me** [comment]
recheck_unit_tests

File: rpc_jobs/setup_statsd.yml [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    name: Setup-StatsD
+    project-type: pipeline
+    concurrent: false
+    properties:
+      - build-discarder:
+          num-to-keep: 30
+    parameters:
+      - rpc_gating_params
+    triggers:
+      - timed: "@daily"
```
Comment (mattt416) [inline rpc_jobs/setup_statsd.yml]: Is this necessary if we already have a push trigger for this?

Comment (odyssey4me) [reply rpc_jobs/setup_statsd.yml]: The daily execution ensures that even if there are no merges going in, the playbook still executes. This ensures that packages are updated and the environment remains configured with the latest keys and such.

File: rpc_jobs/setup_statsd.yml [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+           - master
+    properties:
+      - build-discarder:
+          num-to-keep: 30
+    triggers:
+        - github
+    builders:
+      - trigger-builds:
+        - project:
+            - "Setup-StatsD"
```
Comment (mattt416) [inline rpc_jobs/setup_statsd.yml]: Cleaner to have a separate merge trigger job for both statsd and graphite, or better to combine them into one?

Comment (odyssey4me) [reply rpc_jobs/setup_statsd.yml]: I've registered a follow up card to do the deduplication: https://rpc-openstack.atlassian.net/browse/RE-1472

File: rpc_jobs/setup_statsd.yml (hunk 1 above)
Comment (mattt416) [reply rpc_jobs/setup_statsd.yml]: Fair. :)

**Comment by odyssey4me** [comment]
@mattt416 Quite right, I guess I could have pulled the card # off it - I didn't see any harm including it given it's an improvement relating to the epic anyway.

**Pull request merged**